webfont/make_woff.py

```diff
@@ -0,0 +1,294 @@
+
+
+class FilterPipeLine():
+    def __init__(self):
+        self.filters = []
+
+    def add_filter(self, filt):
+        self.filters.append(filt)
+
+    def remove_filter(self, filt):
+        self.filters.remove(filt)
+
+    def feed(self, codepoint):
+        """Feed codepoint or unicode character."""
+        if not isinstance(codepoint, int):
+            codepoint = ord(codepoint)
+        for filt in self.filters:
+            if filt.feed(codepoint):
+                break
+        else:
+            raise MissingGlyphError(codepoint)
+
+
+class BaseFilter():
+    def feed(self, codepoint):
+        """Let the filter handle the codepoint.
+
+        Returning True means that the codepoint was handled subsequent filters
+        will be skipped.
+
+        """
+        raise NotImplementedError
+
+
+class FontFilter(BaseFilter):
+    def __init__(self, font):
+        self.font = font
+
+    def feed(self, codepoint):
+        return self.font.add_unicode(codepoint)
+
+
+class UniqueFilter(BaseFilter):
+    """Skips already used code points.
+
+    Should be the first one.
+    """
+    def __init__(self):
+        self.fed = set()
+
+    def feed(self, codepoint):
+        if codepoint in self.fed:
+            return True
+        else:
+            self.fed.add(codepoint)
+            return False
+
+
+class ExcludeFilter(BaseFilter):
+    def __init__(self):
+        self.exclude_ranges = []
+
+    def exclude_range(self, unicode_range):
+        self.exclude_ranges.append(unicode_range)
+
+    def feed(self, codepoint):
+        for range_ in self.exclude_ranges:
+            if codepoint in range_:
+                return True
+            else:
+                return False
+
+
+def main():
+    argc = len(argv)
+    if argc == 1:
+        print(_USAGE)
+        exit(0)
+    elif argc == 2:
+        if argv[1] in ['-h', '--help']:
+            print(_USAGE)
+            exit(0)
+        else:
+            try:
+                configpath = argv[1]
+                with open(configpath) as configfile:
+                    config = json.load(configfile)
+            except FileNotFoundError as err:
+                error(f'Config file "{configpath}" not found: {err.strerror}')
+                exit(1)
+            except OSError as err:
+                error(f'Error reading "{configpath}": {err.strerror}')
+                exit(1)
+            except json.JSONDecodeError as err:
+                error(f'JSON error in "{configpath}": {err}')
+                exit(1)
+            configdir = Path(argv[1]).parent
+
+    ft_logger = logging.getLogger('fontTools.subset')
+
+    def silence_unknown_subset_table(record):
+        return record.msg != "%s NOT subset; don't know how to subset; dropped"
+
+    ft_logger.addFilter(silence_unknown_subset_table)
+
+    pipeline = FilterPipeLine()
+
+    uniq_filt = UniqueFilter()
+    pipeline.add_filter(uniq_filt)
+
+    excl_filt = ExcludeFilter()
```

Comment:
The ASCII range takes up near-as-makes-no-difference nothing compared to the rest of the data that gets loaded, and having the Hanazono-specific latin characters in the font means that stretches that end up being marked as "use Japanese typesetting, even for the latin bits" will look better later.

Let's take out the exclusion option.